StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/VerticalPaymentMethodListViewController.swift

```diff
@@ -230,6 +230,21 @@ enum VerticalPaymentMethodListSelection: Equatable {
             return false
         }
     }
+    
+    func hash(into hasher: inout Hasher) {
+        switch self {
+        case .new(let paymentMethodType):
+            hasher.combine(0)
+            hasher.combine(paymentMethodType.identifier)
+        case .saved(let paymentMethod):
+            hasher.combine(1)
+            hasher.combine(paymentMethod.stripeId)
+        case .applePay:
+            hasher.combine(2)
+        case .link:
+            hasher.combine(3)
+        }
+    }
```

Comment:
Is this needed? I can compile without this function?

Reply:
Yeah it compiles without this function, but the auto synthesized version is a black box. Given we have non trivial associated values such as `PaymentSheet.PaymentMethodType` and `STPPaymentMethod`, I'd prefer to define how and what we hash here rather than hoping the auto synthesized one is what we want.

Similarly we could remove the `static func == (lhs: VerticalPaymentMethodListSelection, rhs: VerticalPaymentMethodListSelection) -> Bool` and keep the `Equatable` conformance and it still compiles.

Reply:
In fact, if you remove this function and run `EmbeddedUITest.testSelection()` it fails!

Reply:
Hmm, I'd argue that unless it is absolutely needed, it should be deleted, so in this case, I think it would be better to delete both functions.

Looking at other examples in the code base, we tend to do it for specific reasons:
  PaymentMethodAvailabilityStatus: Equtable -- needed because we don't care about ordering.
  EmbeddedPaymentElement.UpdateResult: Equatable -- needed because underlying types are not Equatable.

Reply:
Bringing this out from Slack convo

```swift
    func testHashSelection() {
        let selection = VerticalPaymentMethodListSelection.saved(paymentMethod: ._testCard())
        let selection1 = VerticalPaymentMethodListSelection.saved(paymentMethod: ._testCard())
        
        // Fails with auto synthesized Hashable implementation, passes with custom
        XCTAssertEqual(selection.hashValue, selection1.hashValue)
    }
    
    func testHashSelection_sameInstance() {
        let testCard: STPPaymentMethod = ._testCard()
        let selection = VerticalPaymentMethodListSelection.saved(paymentMethod: testCard)
        let selection1 = VerticalPaymentMethodListSelection.saved(paymentMethod: testCard)
        
        // Passes with auto synthesized Hashable implementation, passes with custom
        XCTAssertEqual(selection.hashValue, selection1.hashValue)
    }
```